scipy/optimize/tests/test_trustregion_krylov.py

```diff
@@ -0,0 +1,103 @@
+"""
+Unit tests for Krylov space trust-region subproblem solver.
+
+To run it in its simplest form::
+  nosetests test_optimize.py
+
+"""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.optimize._trlib import (TRLIBQuadraticSubproblem)
+from numpy.testing import (TestCase, assert_, assert_array_equal,
+                           assert_almost_equal,
+                           assert_equal, assert_array_almost_equal,
+                           assert_array_less, run_module_suite)
+
+KrylovQP = TRLIBQuadraticSubproblem(tol_rel_i=1e-8, tol_rel_b=1e-6)
+
+class TestKrylovQuadraticSubproblem(TestCase):
+
+    def test_for_the_easy_case(self):
```

Comment:
You are absolutely right, those test are not very meaningful from an optimization point of view.

When I sugested it I was thinking in terms of  the tests I used for the ``trust-exact`` method. The ``trust-exact`` method can, in theory, escape from saddle points, and that is why I have implemented those test when I was implementing it. But since all trust_region methods in scipy stops when the gradient is sufficiently low, this capability is not being currently used.